docs/reference/sql/limitations.asciidoc

```diff
@@ -3,6 +3,46 @@
 [[sql-limitations]]
 == SQL Limitations
 
+[float]
+[[large-parsing-trees]]
+=== Large queries may throw `ParsingExpection`
+
+{es-sql} uses https://www.antlr.org/[`ANTLR`] to create https://en.wikipedia.org/wiki/Abstract_syntax_tree[abstract syntax trees]
+from a query as a first step before its analysis, planning and execution. As a consequence certain types of queries or
+expressions inside them can generate abstract syntax trees of large depth which will exhaust the JVM's stack and will
+trigger a `StackOverflowException`. {es-sql}'s parser will handle the `StackOverflowException` and throw a user facing
+`ParsingException` with a descriptive error message. Below you can find some examples of queries that could trigger
+this.
+
+- Large arithmetic operator expressions
```

Comment:
I'm not sure the examples of large trees would help - these are suppose to be corner-cases and I would expect it to be obvious (thousands of entries). Offering such a predominant section in the docs, makes it look like a common occurrence which is not the case.